lib/bookmarks.js

```diff
@@ -0,0 +1,39 @@
+import bookmarkModel from '../models';
+
+const { bookmarks, articles, users } = bookmarkModel;
+
+/** @description validate bookmarks
+   * @param {integer} userId is the request parameter
+   * @param {integer} articleId is the response parameter
+   * @return {object} the response object
+   * @public
+   */
+const validateBookmarks = async (userId, articleId) => {
+  try {
+    const user = await users.findOne({
+      where: {
+        id: userId
+      }
+    });
+    const article = await articles.findOne({
+      where: {
+        id: articleId
+      }
+    });
+    const isAvalaible = await bookmarks.findOne({
+      where: {
+        userId,
+        articleId
+      }
+    });
+
+    if (!article || isAvalaible || !user) {
+      return 'failure, article cannot be bookmarked';
+    }
+    return 'success';
```

Comment:
Why return the String `success`?

Reply:
Success is returned on the function saying that it passes all validations to create a bookmark, I did not want to send the res directly to the user from the lib folder..on line 14 in the middleware folder bookmarksValidate.js I checked if the return statement is Success, if not success then I then send a response back to the user  @Chidi3b3r3

Reply:
Please use booleans for binary operations. If success `true` otherwise `false`. Strings are difficult to manage and more prone to errors